### PR TITLE
Use updated binutils-vc4 opcodes for vc4 disasm and provide framework for vc4 asm

### PR DESCRIPTION
**Pull request by zamaudio**

### CODE REVIEW

**Comment by zamaudio** [comment]
This is currently a copy-paste from the binutils-vc4 by puppeh.  I haven't even read much of the code so I don't know about its quality. I just made it compile and the disassembler currently functions.

**Comment by radare** [comment]
Do you think this should go into master or just as a plugin shipped in r2e?

**Comment by zamaudio** [comment]
What is r2e?

**Comment by zamaudio** [comment]
I think raspberry pi is quite popular for many users, so being able to disassemble the firmware is fun for a lot of people.

**Comment by zamaudio** [comment]
The assembler support is also bundled with this patch, I just need to write a wrapper for a plugin to use the assembler.

**Comment by radare** [comment]
yeah, i wanted it for this too. i was thinking in fuzzing it and properly reviewing the code + fixing bugs with clang-analyzer/coverity and such before that. because the current code is full of vulnerabilities. (didnt checked your  pr yet)

**Comment by radare** [comment]
I think that initially we should distribute this in r2e. if someone wants to use it just r2pm -i vc4.. at least we will not make r2 bigger and buggier.

**Comment by zamaudio** [comment]
Ok sure

**Comment by radare** [comment]
I gave you commit access to radare2-extras, feel free to do the transition :)

**Comment by radare** [comment]
ping

**Comment by zamaudio** [comment]
@rarare Can you do review here first?  I don't want to merge 15k LOC into r2e its too much.

File: libr/asm/arch/vc4/libvc4/vc4_arch.c [hunk 1]
```diff
@@ -704,129 +731,74 @@ static void vc4_add_opcode(struct vc4_info *info, struct vc4_opcode *op)
...
-	if (!strchr(format, '\"')) {
-		fprintf(stderr, "No closing \"! [%s] \n", format);
-		abort();
-	}
-	*strchr(format, '"') = 0;
-
-	op = (struct vc4_opcode *)calloc(1, sizeof(struct vc4_opcode));
-	op->format = format;
-	op->length = nibs / 4;
+static int scan_format(char *d, const char **pp)
```
Comment (radare) [inline libr/asm/arch/vc4/libvc4/vc4_arch.c]: wrong indent, braces at right

**Comment by radare** [comment]
its hard to review code that mixes gnu style and our style depending on the source of it, i would prefer to merge in there and cleanup after commit. it will be easier for all us.

Also, i dont think we should review this much more if merged into r2e because its not a core functionality, it have 0 tests, it includes tons of lines from other projects and it doesnt follows any static coding style

**Comment by radare** [comment]
ping

**Comment by radare** [comment]
PING

**Comment by radare** [comment]
merged. i will now move all this stuff into the r2e repo and remove the branch

**Comment by radare** [comment]
done in https://github.com/radare/radare2-extras/commit/7500101a859127659f3256521a61ed937cb90ff6

**Pull request closed without merging**